internal/config/config.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/alecthomas/kingpin/v2"
+	"gopkg.in/yaml.v3"
+)
+
+// Config represents the complete application configuration
+type (
+	Log struct {
+		Level  string `yaml:"level"`
+		Format string `yaml:"format"`
```

Comment:
should config default values be supported using go struct tags? or ALL future config default values should be defined in `DefaultConfig()` only?

Reply:
> should config default values be supported using go struct tags?

Not sure if that works ? https://pkg.go.dev/gopkg.in/yaml.v3#section-readme 

> ALL future config default values should be defined in DefaultConfig() only?

yes, that's the idea, put all Defaults (including those that may need to be computed) in this function.

Reply:
> Not sure if that works ? https://pkg.go.dev/gopkg.in/yaml.v3#section-readme

we will need something more than just yaml package.